### ISSUE TITLE
Reference to GitHub account to access argocd install is misleading

### ISSUE THREAD

**Issue by jsoref** [argoproj/argo-cd#4837]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [ ] I've pasted the output of `argocd version`.

**Describe the bug**

The docs say you will need to `sign-in` (n.b. ArgoCD uses "Log in" as its button) with your `github` (n.b. this should be `GitHub`) account, but:
1. It's possible to access the system w/o logging in
1. Logging in w/ my GitHub account gave me an access refused error:
   > Failed to authenticate: github: user "jsoref" not in required orgs or teams

**To Reproduce**

1. Visit https://argoproj.github.io/argo-cd/operator-manual/declarative-setup/
1. Scroll to the bottom
1. See:
   > You will need to sign-in using your github account to get access to https://cd.apps.argoproj.io
1. Click https://cd.apps.argoproj.io
1. Click `Log in`
1. Click `Log in via github`
1. Authorize my GitHub account

**Expected behavior**

Docs shouldn't be misleading.
* It should indicate you don't need to log in to access
* If it's going to mention logging in, it should indicate who will be able to log in

**Screenshots**
**Version**
**Logs**

**Comment by jessesuen** [comment]
> You will need to sign-in using your github account to get access to https://cd.apps.argoproj.io

yes, this was written back when we did not have anonymous access feature.

**Comment by wojtekidd** [comment]
Can I take this one for a spin @jessesuen ?
